plugin/keyring_vault/vault_memory.h

```diff
@@ -0,0 +1,47 @@
+#ifndef MYSQL_VAULT_MEMORY
+#define MYSQL_VAULT_MEMORY
+
+#include <my_global.h>
+#include <string.h>
+#include "keyring_memory.h"
+
+namespace keyring
+{
+#if !defined(HAVE_STDCXX11) && !defined(WIN32)
+  inline void memset_s(void *dest, size_t dest_max, int c, size_t n)
+  {
+    volatile unsigned char *p = reinterpret_cast<unsigned char*>(dest);
+    while (dest_max-- && n--) {
+      *p++ = c;
+    }
+  } 
+#endif
+
+  template <class T> class Secure_allocator : public std::allocator<T>
```

Comment:
I agree, this is not a serious issue. Just a good design principle - do not use public inheritance for extending/modifying functionality of classes which were never supposed to be inherited.